File: comprehensive_python_syntax.py
#!/usr/bin/env python3
"""
Comprehensive Python syntax demonstration file
Contains all major Python language features
"""

# Import statements
import sys
import os
from typing import List, Dict, Optional, Union, Any, Callable, Iterator, Generator
from abc import ABC, abstractmethod
from dataclasses import dataclass
from enum import Enum
import asyncio
import threading
import time
import json

# Constants and variables
CONSTANT_VALUE = 42
PI = 3.14159
STRING_CONSTANT = "Hello, World!"

# Basic data types
integer_var = 42
float_var = 3.14
string_var = "Hello Python"
bool_var = True
none_var = None

# Collection types
list_var = [1, 2, 3, 4, 5]
tuple_var = (1, 2, 3, 4, 5)
set_var = {1, 2, 3, 4, 5}
dict_var = {"key1": "value1", "key2": "value2"}

# List comprehensions
squares = [x**2 for x in range(10)]
even_squares = [x**2 for x in range(10) if x % 2 == 0]
matrix = [[i*j for j in range(3)] for i in range(3)]

# Dictionary comprehensions
square_dict = {x: x**2 for x in range(5)}

# Set comprehensions
unique_squares = {x**2 for x in range(10)}

# Generator expressions
gen = (x**2 for x in range(10))

# Functions
def basic_function(x: int, y: str = "default") -> str:
    """Basic function with type hints"""
    return f"{y}: {x}"

def function_with_args(*args, **kwargs):
    """Function with variable arguments"""
    print(f"args: {args}")
    print(f"kwargs: {kwargs}")

def generator_function(n: int) -> Generator[int, None, None]:
    """Generator function"""
    for i in range(n):
        yield i

# Lambda functions
lambda_func = lambda x, y: x + y
lambda_with_condition = lambda x: "even" if x % 2 == 0 else "odd"

# Class definitions
class BasicClass:
    """Basic class with various features"""
    
    class_variable = "I'm a class variable"
    
    def __init__(self, name: str, age: int = 0):
        self.name = name
        self.age = age
        self._private_var = "private"
    
    def instance_method(self):
        return f"Hello, I'm {self.name}"
    
    @classmethod
    def class_method(cls):
        return f"Class method called on {cls.__name__}"
    
    @staticmethod
    def static_method():
        return "Static method called"
    
    @property
    def name_property(self):
        return self.name.upper()
    
    def __str__(self):
        return f"BasicClass(name='{self.name}', age={self.age})"
    
    def __repr__(self):
        return self.__str__()

# Inheritance
class ChildClass(BasicClass):
    """Child class demonstrating inheritance"""
    
    def __init__(self, name: str, age: int, specialty: str):
        super().__init__(name, age)
        self.specialty = specialty
    
    def child_method(self):
        return f"{self.name} specializes in {self.specialty}"

# Abstract base class
class AbstractShape(ABC):
    """Abstract base class"""
    
    @abstractmethod
    def area(self) -> float:
        pass
    
    @abstractmethod
    def perimeter(self) -> float:
        pass

class Rectangle(AbstractShape):
    """Concrete implementation of abstract class"""
    
    def __init__(self, width: float, height: float):
        self.width = width
        self.height = height
    
    def area(self) -> float:
        return self.width * self.height
    
    def perimeter(self) -> float:
        return 2 * (self.width + self.height)

# Data classes
@dataclass
class Person:
    """Data class example"""
    name: str
    age: int
    email: Optional[str] = None

# Enums
class Color(Enum):
    """Enum example"""
    RED = 1
    GREEN = 2
    BLUE = 3

# Exception handling
def demonstrate_exceptions():
    """Demonstrate exception handling"""
    try:
        result = 10 / 0
    except ZeroDivisionError as e:
        print(f"Caught ZeroDivisionError: {e}")
    except Exception as e:
        print(f"Caught general exception: {e}")
    else:
        print("No exception occurred")
    finally:
        print("Finally block executed")

# Context managers
class CustomContextManager:
    """Custom context manager"""
    
    def __enter__(self):
        print("Entering context")
        return self
    
    def __exit__(self, exc_type, exc_val, exc_tb):
        print("Exiting context")
        return False

def use_context_manager():
    """Use context manager"""
    with CustomContextManager() as cm:
        print("Inside context")

# Decorators
def simple_decorator(func):
    """Simple decorator"""
    def wrapper(*args, **kwargs):
        print(f"Calling {func.__name__}")
        result = func(*args, **kwargs)
        print(f"Finished calling {func.__name__}")
        return result
    return wrapper

@simple_decorator
def decorated_function():
    """Function with decorator"""
    print("Decorated function executed")

# Async functions
async def async_function():
    """Async function"""
    print("Async function started")
    await asyncio.sleep(0.1)
    print("Async function completed")
    return "async result"

async def async_generator():
    """Async generator"""
    for i in range(3):
        await asyncio.sleep(0.1)
        yield i

# Threading
def worker_function(name: str, delay: float):
    """Worker function for threading"""
    print(f"Worker {name} started")
    time.sleep(delay)
    print(f"Worker {name} finished")

def demonstrate_threading():
    """Demonstrate threading"""
    threads = []
    for i in range(3):
        thread = threading.Thread(target=worker_function, args=(f"Thread-{i}", 0.1))
        threads.append(thread)
        thread.start()
    
    for thread in threads:
        thread.join()

# File I/O operations
def file_operations():
    """Demonstrate file operations"""
    # Write to file
    with open("test_file.txt", "w") as f:
        f.write("Hello, file!\n")
        f.write("Second line\n")
    
    # Read from file
    with open("test_file.txt", "r") as f:
        content = f.read()
        print(f"File content: {content}")
    
    # Clean up
    os.remove("test_file.txt")

# JSON operations
def json_operations():
    """Demonstrate JSON operations"""
    data = {
        "name": "John",
        "age": 30,
        "hobbies": ["reading", "coding", "gaming"]
    }
    
    # Convert to JSON
    json_str = json.dumps(data, indent=2)
    print(f"JSON string: {json_str}")
    
    # Parse JSON
    parsed_data = json.loads(json_str)
    print(f"Parsed data: {parsed_data}")

# Regular expressions
import re

def regex_operations():
    """Demonstrate regular expressions"""
    pattern = r"\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b"
    text = "Contact us at support@example.com or sales@company.org"
    
    matches = re.findall(pattern, text)
    print(f"Email matches: {matches}")

# Functional programming
def functional_programming():
    """Demonstrate functional programming concepts"""
    numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
    
    # Map
    squared = list(map(lambda x: x**2, numbers))
    print(f"Squared numbers: {squared}")
    
    # Filter
    evens = list(filter(lambda x: x % 2 == 0, numbers))
    print(f"Even numbers: {evens}")
    
    # Reduce
    from functools import reduce
    sum_all = reduce(lambda x, y: x + y, numbers)
    print(f"Sum of all numbers: {sum_all}")

# Operator overloading
class Vector:
    """Class with operator overloading"""
    
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y
    
    def __add__(self, other):
        return Vector(self.x + other.x, self.y + other.y)
    
    def __mul__(self, scalar: float):
        return Vector(self.x * scalar, self.y * scalar)
    
    def __str__(self):
        return f"Vector({self.x}, {self.y})"

# Multiple inheritance
class MixinA:
    def method_a(self):
        return "MixinA method"

class MixinB:
    def method_b(self):
        return "MixinB method"

class MultipleInheritance(MixinA, MixinB):
    pass

# Metaclasses
class MyMeta(type):
    """Custom metaclass"""
    
    def __new__(cls, name, bases, namespace):
        print(f"Creating class {name}")
        return super().__new__(cls, name, bases, namespace)

class MetaClassExample(metaclass=MyMeta):
    pass

# Global and nonlocal
def demonstrate_scope():
    """Demonstrate global and nonlocal scope"""
    global_var = "I'm global"
    
    def outer_function():
        outer_var = "I'm outer"
        
        def inner_function():
            nonlocal outer_var
            inner_var = "I'm inner"
            outer_var = "Modified by inner"
            return inner_var
        
        result = inner_function()
        print(f"Outer var after inner call: {outer_var}")
        return result
    
    return outer_function()

# Walrus operator (Python 3.8+)
def walrus_operator_example():
    """Demonstrate walrus operator"""
    numbers = [1, 2, 3, 4, 5]
    
    # Traditional way
    n = len(numbers)
    if n > 3:
        print(f"List has {n} elements")
    
    # With walrus operator
    if (n := len(numbers)) > 3:
        print(f"List has {n} elements (walrus)")

# Pattern matching (Python 3.10+)
def pattern_matching_example(value):
    """Demonstrate pattern matching"""
    match value:
        case 1:
            return "One"
        case 2:
            return "Two"
        case 3:
            return "Three"
        case [x, y]:
            return f"List with {x} and {y}"
        case {"key": key_value}:
            return f"Dict with key: {key_value}"
        case _:
            return "Something else"

# Main execution function
def main():
    """Main function to demonstrate all features"""
    print("=== Comprehensive Python Syntax Demo ===")
    
    # Basic operations
    print(f"Basic variables: {integer_var}, {float_var}, {string_var}")
    print(f"Collections: {list_var}, {dict_var}")
    print(f"List comprehension: {squares}")
    print(f"Function result: {basic_function(42, 'test')}")
    
    # Class operations
    obj = BasicClass("Test Object", 25)
    print(f"Object: {obj}")
    print(f"Instance method: {obj.instance_method()}")
    print(f"Class method: {BasicClass.class_method()}")
    print(f"Static method: {BasicClass.static_method()}")
    
    # Inheritance
    child = ChildClass("Child", 10, "Python")
    print(f"Child: {child.child_method()}")
    
    # Abstract class
    rect = Rectangle(5, 3)
    print(f"Rectangle area: {rect.area()}")
    print(f"Rectangle perimeter: {rect.perimeter()}")
    
    # Data class
    person = Person("Alice", 30, "alice@example.com")
    print(f"Person: {person}")
    
    # Enum
    print(f"Color enum: {Color.RED}")
    
    # Exception handling
    demonstrate_exceptions()
    
    # Context manager
    use_context_manager()
    
    # Decorator
    decorated_function()
    
    # Threading
    demonstrate_threading()
    
    # File operations
    file_operations()
    
    # JSON operations
    json_operations()
    
    # Regex operations
    regex_operations()
    
    # Functional programming
    functional_programming()
    
    # Operator overloading
    v1 = Vector(1, 2)
    v2 = Vector(3, 4)
    v3 = v1 + v2
    print(f"Vector addition: {v3}")
    
    # Multiple inheritance
    multi = MultipleInheritance()
    print(f"Multiple inheritance: {multi.method_a()}, {multi.method_b()}")
    
    # Scope demonstration
    scope_result = demonstrate_scope()
    print(f"Scope result: {scope_result}")
    
    # Walrus operator
    walrus_operator_example()
    
    # Pattern matching
    print(f"Pattern matching 1: {pattern_matching_example(1)}")
    print(f"Pattern matching [1,2]: {pattern_matching_example([1, 2])}")
    print(f"Pattern matching dict: {pattern_matching_example({'key': 'value'})}")
    
    # Async operations (run in event loop)
    async def run_async():
        result = await async_function()
        print(f"Async result: {result}")
        
        async for item in async_generator():
            print(f"Async generator item: {item}")
    
    asyncio.run(run_async())
    
    print("=== Demo completed ===")

if __name__ == "__main__":
    main()

File: comprehensive_python_syntax.cpp
#include <iostream>
#include <string>
#include <vector>
#include <optional>
#include <unordered_map>
#include <tuple>
#include <utility>
#include <type_traits>
#include <functional>
#include <sstream>
#include <iomanip>
#include <cmath>
#include <algorithm>
#include <typeinfo>
#include <cstring>

using namespace std;

// helpers
template<typename T> static inline std::ostream& operator<<(std::ostream& os, const std::vector<T>& v);
template<typename T> static inline std::string to_str(const T& v){ std::ostringstream os; os<<v; return os.str(); }
static inline std::string to_str(const std::string& s){ return s; }
static inline std::string to_str(const char* s){ return std::string(s); }
static inline std::string to_str(double v){ std::ostringstream os; os.setf(std::ios::fixed); os<<std::setprecision(1)<<v; return os.str(); }
static inline std::string to_str(float v){ std::ostringstream os; os.setf(std::ios::fixed); os<<std::setprecision(1)<<v; return os.str(); }
template<typename T> static inline int to_int(const T& v){ if constexpr (std::is_convertible_v<T,std::string>) return std::stoi(v); else return static_cast<int>(v); }
template<typename T> static inline float to_float(const T& v){ if constexpr (std::is_convertible_v<T,std::string>) return std::stof(v); else return static_cast<float>(v); }
template<typename C> static inline int sum(const C& c){ int s=0; for(const auto& e: c) s += e; return s; }
template<typename T> static inline int sum(std::initializer_list<T> c){ int s=0; for(const auto& e: c) s += e; return s; }
template<typename T> static inline size_t len(const T& c){ return c.size(); }
static inline size_t len(const char* s){ return std::char_traits<char>::length(s); }
template<typename K, typename V> static inline std::string to_str(const std::unordered_map<K,V>&){ return std::string("{...}"); }
// minimal ABC base to support 'class X : public ABC'
struct ABC {};
// minimal math namespace replacement for 'math.pi'
struct __fluxus_math { static constexpr double pi = 3.141592653589793; static inline double sqrt(double x){ return std::sqrt(x); } } math;
// minimal sys module stub
namespace sys { static std::vector<std::string> argv; }
namespace asyncio { template<typename F> static inline void run(F f){ f(); } }

// vector to string conversion for printing
template<typename T>
static inline std::string vec_to_str(const std::vector<T>& v) {
    std::ostringstream os;
    os << "[";
    for (size_t i = 0; i < v.size(); ++i) {
        if (i > 0) os << ", ";
        if constexpr (std::is_same_v<T, std::string>) {
            os << "'" << v[i] << "'";
        } else {
            os << v[i];
        }
    }
    os << "]";
    return os.str();
}

// nested vector to string
template<typename T>
static inline std::string vec_to_str(const std::vector<std::vector<T>>& v) {
    std::ostringstream os;
    os << "[";
    for (size_t i = 0; i < v.size(); ++i) {
        if (i > 0) os << ", ";
        os << vec_to_str(v[i]);
    }
    os << "]";
    return os.str();
}

// stream operator for vectors
template<typename T>
static inline std::ostream& operator<<(std::ostream& os, const std::vector<T>& v) {
    os << vec_to_str(v);
    return os;
}
template<typename K, typename V> static inline std::ostream& operator<<(std::ostream& os, const std::unordered_map<K,V>&){ os << "{...}"; return os; }
template<typename K, typename V> static inline auto py_items(const std::unordered_map<K,V>& m){ std::vector<std::pair<K,V>> v; v.reserve(m.size()); for(const auto& kv: m){ v.emplace_back(kv.first, kv.second);} return v; }


class BasicClass {
    public:
        BasicClass(std::string name: str, int age: int = 0) {
            this->name = name;
            this->age = age;
            this->_private_var = "private";
        }
        std::string instance_method() {
    return to_str("Hello, I'm ") + to_str(this->name);
}


        std::string class_method(int cls) {
    return to_str("Class method called on ") + to_str(cls);
}


        std::string static_method() {
    return "Static method called";
}


        std::string name_property() {
    return this->name.upper();
}


        std::string __str__() {
    return to_str("BasicClass(name='") + to_str(this->name) + to_str("', age=") + to_str(this->age) + to_str(")");
}


        int __repr__() {
    return this->__str__();
}


    public:
        std::string name = "";
        int age = 0;
        std::string _private_var = "";
};


class ChildClass : public BasicClass {
    public:
        ChildClass(std::string name: str, int age: int, std::string specialty: str) : BasicClass(name, age) {
            this->specialty = specialty;
        }
        std::string child_method() {
    return to_str(this->name) + to_str(" specializes in ") + to_str(this->specialty);
}


    public:
        int specialty = 0;
};


class AbstractShape : public ABC {
    public:
        template<typename... Args> AbstractShape(Args... args) : ABC(args...) {}
        AbstractShape() = default;
        void area() {
    /* pass */
}


        void perimeter() {
    /* pass */
}


    public:
        // generated placeholder fields
};


class Rectangle : public AbstractShape {
    public:
        Rectangle(int width: float, int height: float) {
            this->width = width;
            this->height = height;
        }
        int area() {
    return this->width * this->height;
}


        int perimeter() {
    return 2 * (this->width + this->height);
}


    public:
        int width = 0;
        int height = 0;
};


class Person {
    public:
        Person() = default;
        void placeholder();
    public:
        // generated placeholder fields
};


class Color : public Enum {
    public:
        template<typename... Args> Color(Args... args) : Enum(args...) {}
        Color() = default;
        void placeholder();
    public:
        // generated placeholder fields
};


class CustomContextManager {
    public:
        CustomContextManager() = default;
        int __enter__() {
    std::cout << "Entering context" << std::endl;
    return self;
}


        bool __exit__(int exc_type, int exc_val, int exc_tb) {
    std::cout << "Exiting context" << std::endl;
    return false;
}


    public:
        // generated placeholder fields
};


class Vector {
    public:
        Vector(int x: float, int y: float) {
            this->x = x;
            this->y = y;
        }
        int __add__(int other) {
    return Vector(this->x + other.x, this->y + other.y);
}


        int __mul__(float_ scalar) {
    return Vector(this->x * scalar, this->y * scalar);
}


        std::string __str__() {
    return to_str("Vector(") + to_str(this->x) + to_str(", ") + to_str(this->y) + to_str(")");
}


    public:
        int x = 0;
        int y = 0;
};


class MixinA {
    public:
        MixinA() = default;
        std::string method_a() {
    return "MixinA method";
}


    public:
        // generated placeholder fields
};


class MixinB {
    public:
        MixinB() = default;
        std::string method_b() {
    return "MixinB method";
}


    public:
        // generated placeholder fields
};


class MultipleInheritance : public MixinA, MixinB {
    public:
        template<typename... Args> MultipleInheritance(Args... args) : MixinA, MixinB(args...) {}
        MultipleInheritance() = default;
        void placeholder();
    public:
        // generated placeholder fields
};


class MyMeta : public type {
    public:
        template<typename... Args> MyMeta(Args... args) : type(args...) {}
        MyMeta() = default;
        int __new__(int cls, std::string name, int bases, int namespace) {
    std::cout << to_str("Creating class ") << to_str(name) << std::endl;
    return this->__new__(cls, name, bases, namespace);
}


    public:
        // generated placeholder fields
};


class MetaClassExample : public metaclass=MyMeta {
    public:
        template<typename... Args> MetaClassExample(Args... args) : metaclass=MyMeta(args...) {}
        MetaClassExample() = default;
        void placeholder();
    public:
        // generated placeholder fields
};


std::string basic_function(int_ x, str = "default" y) {
    return to_str(y) + to_str(": ") + to_str(x);
}


void function_with_args(std::initializer_list<int> args, std::initializer_list<int> *kwargs) {
    std::cout << to_str("args: ") << to_str(args) << std::endl;
    std::cout << to_str("kwargs: ") << to_str(kwargs) << std::endl;
}

template<typename... Ts>
auto function_with_args(Ts... xs){ return function_with_args(std::initializer_list<int>{ static_cast<int>(xs)... }); }


void generator_function(int_ n) {
    for (int i = 0; i < n; ++i) {
        /* yield */
    }
}


void demonstrate_exceptions() {
    bool __fluxus_exc=false;
    {
        auto result = 0;
if ((0) == 0) { __fluxus_exc = true; result = 0; }
else { result = ((1.0*(10))/(0)); }

    }
    if (__fluxus_exc) { auto e = 0;
        std::cout << to_str("Caught ZeroDivisionError: ") << to_str(e) << std::endl;
    }
    if (__fluxus_exc) { auto e = 0;
        std::cout << to_str("Caught general exception: ") << to_str(e) << std::endl;
    }
    else {
        std::cout << "No exception occurred" << std::endl;
    }
    {
        std::cout << "Finally block executed" << std::endl;
    }
}


void use_context_manager() {
    {
        std::cout << "Inside context" << std::endl;
    }
}


std::function<int()> simple_decorator(std::function<int()> func) {
    std::function<int()> wrapper = [&]() {
        std::cout << to_str("Calling ") << to_str(func) << std::endl;
        auto result = func(*args, **kwargs);
        std::cout << to_str("Finished calling ") << to_str(func) << std::endl;
        return result;
    };
    return wrapper;
}


void decorated_function__impl() {
    std::cout << "Decorated function executed" << std::endl;
}

void decorated_function() {
    std::cout << "Before function call" << std::endl;
    auto __decor_res = decorated_function__impl();
    std::cout << "After function call" << std::endl;
    return __decor_res;
}


void worker_function(str name, float_ delay) {
    std::cout << to_str("Worker ") << to_str(name) << to_str(" started") << std::endl;
    time.sleep(delay);
    std::cout << to_str("Worker ") << to_str(name) << to_str(" finished") << std::endl;
}


void demonstrate_threading() {
    auto threads = std::vector<int>{};
    for (int i = 0; i < 3; ++i) {
        auto thread = threading.Thread(target=worker_function, args=(f"Thread-{i}", 0.1));
        threads.push_back(thread);
        thread.start();
    }
    for (auto thread : threads) {
        thread.join();
    }
}


void file_operations() {
    {
        f.write("Hello, file!\n");
        f.write("Second line\n");
    }
    {
        auto content = f.read();
        std::cout << to_str("File content: ") << to_str(content) << std::endl;
    }
    os.remove("test_file.txt");
}


void json_operations() {
    auto data = {;
        "name": "John",;
        "age": 30,;
        "hobbies": std::vector<std::string>{"reading", "coding", "gaming"};
    }
    };
    auto json_str = json.dumps(data, indent=2);
    std::cout << to_str("JSON string: ") << to_str(json_str) << std::endl;
    auto parsed_data = json.loads(json_str);
    std::cout << to_str("Parsed data: ") << to_str(parsed_data) << std::endl;
}


void regex_operations() {
    auto pattern = @std::vector<decltype(A-Za-z0-9.-)>{A-Za-z0-9.-}+\.std::vector<decltype(A-Z|a-z)>{A-Z|a-z}{2,}\b";
    auto text = "Contact us at support@example.com or sales@company.org";
    auto matches = re.findall(pattern, text);
    std::cout << to_str("Email matches: ") << to_str(matches) << std::endl;
}


void functional_programming() {
    auto numbers = std::vector<decltype(1)>{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    auto squared = list(map(lambda x: x**2, numbers));
    std::cout << to_str("Squared numbers: ") << to_str(squared) << std::endl;
    auto evens = list(filter(lambda x: x % 2 == 0, numbers));
    std::cout << to_str("Even numbers: ") << to_str(evens) << std::endl;
    auto sum_all = reduce(lambda x, y: x + y, numbers);
    std::cout << to_str("Sum of all numbers: ") << to_str(sum_all) << std::endl;
}


std::function<int()> demonstrate_scope() {
    auto global_var = "I'm global";
    std::function<int()> outer_function = [&]() {
        auto outer_var = "I'm outer";
        std::function<int()> inner_function = [&]() {
            nonlocal outer_var;
            auto inner_var = "I'm inner";
            auto outer_var = "Modified by inner";
            return inner_var;
        };
        auto result = inner_function();
        std::cout << to_str("Outer var after inner call: ") << to_str(outer_var) << std::endl;
        return result;
    };
    return outer_function();
}


void walrus_operator_example() {
    auto numbers = std::vector<decltype(1)>{1, 2, 3, 4, 5};
    auto n = len(numbers);
    if (n > 3) {
        std::cout << to_str("List has ") << to_str(n) << to_str(" elements") << std::endl;
    }
    if ((n := len(numbers)) > 3) {
        std::cout << to_str("List has ") << to_str(n) << to_str(" elements (walrus)") << std::endl;
    }
}


std::string pattern_matching_example(int value) {
    match value:;
        case 1:;
            return "One";
        }
        case 2:;
            return "Two";
        }
        case 3:;
            return "Three";
        }
        case std::vector<decltype(x)>{x, y}:;
            return to_str("List with ") + to_str(x) + to_str(" and ") + to_str(y);
        }
        case {"key": key_value}:;
            return to_str("Dict with key: ") + to_str(key_value);
        }
        case _:;
            return "Something else";
        }
    }
}


int main() {
    std::cout << "=== Comprehensive Python Syntax Demo ===" << std::endl;
    std::cout << to_str("Basic variables: ") << to_str(integer_var) << to_str(", ") << to_str(float_var) << to_str(", ") << to_str(string_var) << std::endl;
    std::cout << to_str("Collections: ") << to_str(list_var) << to_str(", ") << to_str(dict_var) << std::endl;
    std::cout << to_str("List comprehension: ") << to_str(squares) << std::endl;
    std::cout << to_str("Function result: ") << to_str(basic_function(42, 'test')) << std::endl;
    auto obj = BasicClass("Test Object", 25);
    std::cout << to_str("Object: ") << to_str(obj) << std::endl;
    std::cout << to_str("Instance method: ") << to_str(obj.instance_method()) << std::endl;
    std::cout << to_str("Class method: ") << to_str(BasicClass.class_method()) << std::endl;
    std::cout << to_str("Static method: ") << to_str(BasicClass.static_method()) << std::endl;
    auto child = ChildClass("Child", 10, "Python");
    std::cout << to_str("Child: ") << to_str(child.child_method()) << std::endl;
    auto rect = Rectangle(5, 3);
    std::cout << to_str("Rectangle area: ") << to_str(rect.area()) << std::endl;
    std::cout << to_str("Rectangle perimeter: ") << to_str(rect.perimeter()) << std::endl;
    auto person = Person("Alice", 30, "alice@example.com");
    std::cout << to_str("Person: ") << to_str(person) << std::endl;
    std::cout << to_str("Color enum: ") << to_str(Color.RED) << std::endl;
    demonstrate_exceptions();
    use_context_manager();
    decorated_function();
    demonstrate_threading();
    file_operations();
    json_operations();
    regex_operations();
    functional_programming();
    auto v1 = Vector(1, 2);
    auto v2 = Vector(3, 4);
    auto v3 = v1 + v2;
    std::cout << to_str("Vector addition: ") << to_str(v3) << std::endl;
    auto multi = MultipleInheritance();
    std::cout << to_str("Multiple inheritance: ") << to_str(multi.method_a()) << to_str(", ") << to_str(multi.method_b()) << std::endl;
    auto scope_result = demonstrate_scope();
    std::cout << to_str("Scope result: ") << to_str(scope_result) << std::endl;
    walrus_operator_example();
    std::cout << to_str("Pattern matching 1: ") << to_str(pattern_matching_example(1)) << std::endl;
    std::cout << to_str("Pattern matching [1,2]: ") << to_str(pattern_matching_example(std::vector<decltype(1)>{1, 2})) << std::endl;
    std::cout << to_str("Pattern matching dict: ") << to_str(pattern_matching_example({'key') << to_str(")}") << std::endl;
    async def run_async():;
        auto result = await async_function();
        std::cout << to_str("Async result: ") << to_str(result) << std::endl;
        0;
            std::cout << to_str("Async generator item: ") << to_str(item) << std::endl;
        }
    }
    asyncio::run([&](){ return run_async(); });
    std::cout << "=== Demo completed ===" << std::endl;
    return 0;
}


